common/src/main/java/com/sequenceiq/cloudbreak/util/password/PasswordGenerator.java

```diff
@@ -0,0 +1,115 @@
+package com.sequenceiq.cloudbreak.util.password;
+
+import java.security.SecureRandom;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import org.bouncycastle.crypto.digests.SHA512Digest;
+import org.bouncycastle.crypto.prng.SP800SecureRandomBuilder;
+
+public class PasswordGenerator {
```

Comment:
I know it has been always a utility class but I would be happy to see this as a component, or just using a facade component for the default generator like:
```
@Component
public class DefaultPasswordenerator {

	  public String generate() {
		  return PasswordGenerator.getDefault().generate();
	  }
}
```
It would make it possible to inject it and mock in unit tests, so we could verify it's called and the return value too without using `PowerMockito`

Reply:
Yes I agree. In my previous implementation it was a component, now with the default static instance I ended up using a  static method to get it but good point, I will create a facade.